kvflowcontroller: eliminate mutex contention

irfansharif opened the pull request:
Fixes #105508.

Release note: None

irfansharif committed:
kvflowcontroller: eliminate mutex contention

Fixes #105508.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106466)
<!-- Reviewable:end -->

irfansharif committed:
[wip] kvflowcontrol: annotate/fix perf regressions

bananabrick committed:
[wip] *: use compaction-instrumented pebble

irfansharif committed:
[wip] admission: add metric for l0 compacted bytes, generated tokens

And add cluster setting to control smoothing alpha for byte token
computations.

Release note: None

irfansharif committed:
[wip] *: use l0-compaction-prioritized pebble

irfansharif committed:
[wip] roachtest: use disk snapshots for clearrange/*

irfansharif committed:
[wip] *: use l0-compaction-prioritized pebble again

irfansharif committed:
[wip] *: prioritize l0 compactions v3

irfansharif committed:
[wip] *: prioritize l0 compactions v4

irfansharif committed:
[wip] *: prioritize l0 compactions v5

irfansharif committed:
[wip] *: prioritize l0 compactions v6

irfansharif committed:
[wip] *: prioritize l0 compactions v7

irfansharif committed:
[wip] *: prioritize l0 compactions v8

irfansharif committed:
[wip] pebble: disable version check

irfansharif committed:
[wip] *: prioritize l0 compactions v9

irfansharif committed:
[wip] *: introduce L0 reduction factor in IO token calculations

irfansharif committed:
[wip] kvserver: stop bypassing follower work

irfansharif committed:
[wip] *: go back to using only compaction-instrumented pebble

irfansharif committed:
[wip] *: experimentally bump L0 priority in the presence of range delets
